lib/ansible/module_utils/network/common/config.py

```diff
@@ -290,6 +290,9 @@ def _diff_line(self, other):
         for item in self.items:
             if item not in other:
                 updates.append(item)
+            # special trailing parents need to be included in diff
+            elif item.line == 'end-policy' or item.line == 'end-set':
```

Comment:
It seems this configuration is specific for a particular platform. Is it possible to sanitize the config indentation in the module code?

Reply:
right now there is no platform specific instance of NetworkConfig. It can be done to inherit this class at iosxr code and overwrite function 'difference' and 'diff_exact' to handle these special commands. will that work ?

Reply:
pros of above approach: no extra check penalty for other OS, no regression on other
con: code duplication and fixes in common code would need to be ported